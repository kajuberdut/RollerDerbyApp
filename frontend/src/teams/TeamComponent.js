import { NavLink } from "react-router-dom";
import "./TeamComponent.css";


/**  
* Card component for teams
*/

function TeamComponent({ team }) {

    /** Render the team component */
      
    return (
      <div key={"TeamComponent-" + team.groupId } className="TeamComponent" style={{width: '77%', height: 'auto', borderRadius: '5px', bakgroundColor: '', paddingBottom: '10px' }}> 
       <NavLink to={`/teams/${team.groupId}`} className="TeamComponent-Link" style={{color: '#555555', textDecoration: 'none'}}>
        <div>
            <div style={{paddingTop: '8px', paddingLeft: '20px', paddingRight: '20px'}} >
            <h4 style={{paddingTop: '15px'}}>{team.name}</h4>
            </div>           
        </div>
        </NavLink>
      </div>
    );
  }
  
  export default TeamComponent
