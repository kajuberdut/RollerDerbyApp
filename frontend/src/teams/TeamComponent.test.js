import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import TeamComponent from "./TeamComponent";
import { BrowserRouter } from "react-router-dom";

// * Passing as of 2/15/24
// npm test TeamComponent.test.js
// must be in frontend directory 


describe("Tests Team Component Page Renders", () => {

    const team = { 
          groupId: 1,
          name: 'Testing',
        }

    test("team component should render", () => {
      render(
        <BrowserRouter>
              <TeamComponent team={team}/>
        </BrowserRouter>,
      );
    
    }); 
  
    it("matches snapshot", function() {
      const { asFragment } = render(
        <BrowserRouter>
              <TeamComponent team={team}/>
        </BrowserRouter>,
      );
        expect(asFragment()).toMatchSnapshot();
    });

    it('displays team name', async () => {
        render(
            <BrowserRouter>
                <TeamComponent team={team}/>
            </BrowserRouter>
        );

        screen.debug()

            expect(screen.getByText('Testing')).toBeInTheDocument();
      });

})

