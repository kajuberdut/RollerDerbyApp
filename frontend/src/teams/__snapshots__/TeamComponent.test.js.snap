// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Team Component Page Renders matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamComponent"
    style="width: 77%; height: auto; border-radius: 5px; padding-bottom: 10px;"
  >
    <a
      class="TeamComponent-Link"
      href="/teams/1"
      style="color: rgb(85, 85, 85); text-decoration: none;"
    >
      <div>
        <div
          style="padding-top: 8px; padding-left: 20px; padding-right: 20px;"
        >
          <h4
            style="padding-top: 15px;"
          >
            Testing
          </h4>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
